packages/mp/src/mp/validate/__init__.py

```diff
@@ -0,0 +1,341 @@
+    configurations: Configurations,
+) -> list[ValidationResults]:
+    """Validate a list of integration group names within a specific marketplace scope.
+
+    Returns:
+        list[ValidationResults]: List contains the Validation results object
+
+    """
+    validation_outputs: list[ValidationResults] = []
+    if groups:
+        pre_build_output: list[ValidationResults] = _process_groups_for_validation(
+            groups, _run_pre_build_validations
+        )
+        validation_outputs.extend(pre_build_output)
+
+        if not configurations.only_pre_build_validations:
+            marketplace.build_groups(groups)
+
+    return validation_outputs
+
+
+def _process_groups_for_validation(
+    groups: Iterable[pathlib.Path],
+    validation_function: ValidationFn,
+) -> list[ValidationResults]:
+    """Iterate through groups and perform pre-build validation on their integrations.
+
+    Returns:
+        list[ValidationResults]: List contains the Validation results object
+
+    """
+    validation_outputs: list[ValidationResults] = []
+    for group_dir in groups:
+        if group_dir.is_dir() and group_dir.exists():
+            group_output: list[ValidationResults] = _run_validations(
+                group_dir.iterdir(), validation_function
+            )
+            validation_outputs.extend(group_output)
+
+    return validation_outputs
+
+
+def _validate_integrations(
+    integrations: Iterable[pathlib.Path],
+    marketplace: Marketplace,
+    configurations: Configurations,
+) -> list[ValidationResults]:
+    """Validate a list of integration names within a specific marketplace scope.
+
+    Returns:
+        list[ValidationResults]: List contains the Validation results object
+
+    """
+    validation_outputs: list[ValidationResults] = []
+    if integrations:
+        pre_build_output: list[ValidationResults] = _run_validations(
+            integrations, _run_pre_build_validations
+        )
+        validation_outputs.extend(pre_build_output)
+
+        if not configurations.only_pre_build_validations:
+            marketplace.build_integrations(integrations)
+
+    return validation_outputs
+
+
+def _run_validations(
+    integration: Iterable[pathlib.Path], validation_function: ValidationFn
+) -> list[ValidationResults]:
+    """Execute pre-build validation checks on a list of integration paths.
+
+    Returns:
+        list[ValidationResults]: List contains the Validation results object
+
+    """
+    paths: Iterator[pathlib.Path] = (
+        p for p in integration if p.exists() and mp.core.file_utils.is_integration(p)
+    )
+    validation_outputs: list[ValidationResults] = []
+
+    processes: int = mp.core.config.get_processes_number()
+    with multiprocessing.Pool(processes=processes) as pool:
+        results = pool.imap_unordered(validation_function, paths)
+        for res in results:
+            if not res.is_success:
+                validation_outputs.append(res)  # noqa: PERF401
+
```

Comment:
Please use a list comprehension

```suggestion
    processes: int = mp.core.config.get_processes_number()
    with multiprocessing.Pool(processes=processes) as pool:
        results = pool.imap_unordered(validation_function, paths)
        validation_outputs: list[ValidationResults] = [r for r in results if not r.is_success]
```